plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/androidx/sqlite/visitor/SQLiteOpenHelperMethodVisitor.kt

```diff
@@ -0,0 +1,63 @@
+package io.sentry.android.gradle.instrumentation.androidx.sqlite.visitor
+
+import io.sentry.android.gradle.instrumentation.MethodContext
+import io.sentry.android.gradle.instrumentation.wrap.Replacement
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Type
+import org.objectweb.asm.commons.AdviceAdapter
+import org.objectweb.asm.commons.Method
+
+class SQLiteOpenHelperMethodVisitor(
+    apiVersion: Int,
+    originalVisitor: MethodVisitor,
+    instrumentableContext: MethodContext
+) : AdviceAdapter(
+    apiVersion,
+    originalVisitor,
+    instrumentableContext.access,
+    instrumentableContext.name,
+    instrumentableContext.descriptor
+) {
+
+    private val replacement = Replacement(
+        "Lio/sentry/android/sqlite/SentrySupportSQLiteOpenHelper;",
+        "create",
+        "(Landroidx/sqlite/db/SupportSQLiteOpenHelper;)Landroidx/sqlite/db/SupportSQLiteOpenHelper;"
+    )
+
+    var alreadyInstrumented = false
+
+    override fun visitMethodInsn(
+        opcode: Int,
+        owner: String?,
+        name: String?,
+        descriptor: String?,
+        isInterface: Boolean
+    ) {
+        if (
```

Comment:
nit: I think there's one uncovered case with this approach, for example:
```kotlin
private val delegate: OpenHelper.Factory

private fun wrap(configuration: OpenHelper.Configuration): OpenHelper {
  return delegate.create(configuration)
}

override fun create(configuration: OpenHelper.Configuration): OpenHelper = wrap(configuration)
```

in this case we'd still instrument it, because `delegate.create(configuration)` is outside the `create` method scope. But I think this is still fine and it's more of an edge case; we can always come back to it if someone reports an issue.

Reply:
Another weird thing - I tried to run it locally and decompile afterwards, and I see this weird `bridge synthetic create` method in `AutoClosingRoomOpenHelperFactory`:
![image](https://github.com/getsentry/sentry-android-gradle-plugin/assets/4999776/e6f283c7-2fcf-4814-b527-85b3b5fbc592)

However, I have no idea where it's coming from, the [implementation](https://github.com/androidx/androidx/blob/0cd9a1b62bab73952022bbd055bcd27efa8240a9/room/room-runtime/src/main/java/androidx/room/AutoClosingRoomOpenHelperFactory.kt) doesn't look much different from the other open helpers they have. So I'm guessing it's still fine, because the other `create` method is properly not instrumented. Maybe it's a bug in the decompiler tool not sure

Reply:
bottom line: nothing to address here, just we should be aware of both of these things :)